router/http/src/test/scala/io/buoyant/router/http/DiscardProxySideEffectsTest.scala

```diff
@@ -0,0 +1,66 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Status, _}
+import com.twitter.util.Future
+import io.buoyant.test.Awaits
+import org.scalatest.FunSuite
+
+class DiscardProxySideEffectsTest extends FunSuite with Awaits {
+
+  def service: Service[Request, Response] = {
+    val svc = Service.mk[Request, Response] { req =>
+      val rsp = Response()
+      rsp.status = Status.Ok
+      rsp.setContentString(req.uri)
+      rsp.headerMap ++= req.headerMap
+      Future.value(rsp)
+    }
+
+    val stk = DiscardProxySideEffects.module.toStack(
+      Stack.Leaf(DiscardProxySideEffects.module.role, ServiceFactory.const(svc))
+    )
+
+    await(
+      stk.make(Stack.Params.empty)()
+    )
+  }
+
+  test("filter rewrites uri and drops Proxy-* headers for proxied requests") {
+    val req = Request(Method.Get, "http://acme.co:8080/foo;matrix=uri;/bar?x=1&y=2")
+    req.host = "acme.co:8080"
```

Comment:
This test should have a different host value so that the test validates that the URI's host value is used.